pennylane/utils.py

```diff
@@ -393,3 +393,54 @@ def expand_vector(vector, original_wires, expanded_wires):
     )
 
     return qml.math.reshape(expanded_tensor, 2**M)
+
+
+def prod_sort(op_list, wire_map: dict = None):
+    """Insertion sort algorithm that sorts a list of product factors by their wire indices, taking
+    into account the operator commutivity.
+
+    Args:
+        op_list (List[.Operator]): list of operators to be sorted
+        wire_map (dict): Dictionary containing the wire values as keys and its indexes as values.
+            Defaults to None.
+
+    Returns:
+        List[.Operator]: sorted list of operators
+    """
+
+    def swappable_ops(op1, op2, wire_map: dict) -> bool:
```

Comment:
Since `swappable_ops` doesn't use the outer-scope namespace, moving it out of `prod_sort` makes it unittest-able.

Reply:
I moved it inside of `prod_sort` because I could not think of any example that needs this function besides `prod_sort`.

Reply:
Done!